src/coreclr/vm/methodtable.cpp

```diff
@@ -4026,6 +4026,391 @@ int MethodTable::GetRiscV64PassStructInRegisterFlags(CORINFO_CLASS_HANDLE cls)
 }
 #endif
 
+#if !defined(DACCESS_COMPILE)
+namespace
+{
+    enum class SwiftPhysicalLoweringTag : uint8_t
```

Comment:
Is there a compelling reason for this to be restricted to `uint8_t`? Since this type isn't used in a transfer type and not expressing any sort of binary contract, I would let the compiler do its job and optimize the size as it see fit.

Reply:
Since this is an `enum class`, the underlying type by default is `int32_t`, not implementation-defined. If I'm going to use this in a memset as per your other comment, I want to keep this as unit8_t width.

Reply:
There is no need for that in this case. The size argument here is unlikely to be observable in this situation. Prove it is observable and let's do it but going with the defaults is the most appropriate course of action.

Reply:
Also, defining the underlying type is [against the Core guidelines](https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#enum7-specify-the-underlying-type-of-an-enumeration-only-when-necessary), unless there it is necessary. In this case, necessary would be observable performance impact, aligning with some transfer type, or binary contract.

Reply:
Here's a GodBolt example showing that it's observable https://godbolt.org/z/f8c5TKdxq.

I'll remove the size specification and multiply `size` by `sizeof(SwiftPhysicalLoweringTag)` so I can remove the explicit underlying type.

Reply:
> Here's a GodBolt example showing that it's observable https://godbolt.org/z/f8c5TKdxq.

Will we be relying on that sort of logic in this case?

> I'll remove the size specification and multiply size by sizeof(SwiftPhysicalLoweringTag) so I can remove the explicit underlying type.

Thank you.

Reply:
I took another look at this in Godbolt after having made the change in the online editor. We can't remove the explicit underlying type, as `memset` always sets every byte (there's no way to specify a stride). So either we change this back to a loop, or we make the enum class 8-bit width. This is a case where the behavior is observable.

Godbolt example: https://godbolt.org/z/ToYf6WMMx